src/Pact/Analyze/Eval/Term.hs

```diff
@@ -457,6 +464,15 @@ evalTerm = \case
     succeeds %= (.&& whetherInPact)
     view aeTrivialGuard
 
+  MkUserGuard _ty _obj _name ->
+    --
+    -- NOTE: for now we just leave this uninterpreted, until we are prepared to
+    -- close over and analyze user guard functions in the future. we do not
+    -- try to resolve two uses of the same name to the same user guard, because
+    -- the guard created is *also* a function of the supplied object.
```

Comment:
is the returned user guard the same if passed the same object?

Reply:
As-is, no. If we wanted to be _slightly_ more precise we _could_ switch to an SFunArray from objects to guards, but since we are already completely ignoring the contents of the function we are turning into a guard, this doesn't seem to buy us anything. I can't think of a use case that would benefit from two different user guards unifying because they had the same input. Also as-is in the code, it's clearer that we are straight-up punting on trying to do the analysis here -- the SFunArray might partially obscure that.

Reply:
I'm more interested in the semantics of pact actual rather than the current analysis implementation

Reply:
Ah -- yeah in pact, if you close over equivalent objects, then if you were to invoke the two user guards in the same spot (recalling that user guards can read from the DB), you'd have the same effect.